### DISCUSSION TITLE
question: Using inpainting to add details to portraits, mask faces

### DISCUSSION THREAD

**Discussion by devzaak** [Acly/krita-ai-diffusion#765] [Q&A] [answered]
Hi,
first off, great work on this plugin
And now for the question: I am trying to use the inpainting to add details to photography images. Mainly as requested by people in photos for privacy reasons. I would like to add things like face-masks, venetian masks etc that fit the general shape of the face but not paint in a new face or change the shape of the head etc. That way edits especially in group photos would be a bit more subtle and not too disruptive.
Is this a bit too much at this point or is there a way to do something like that?
Thank you

**Comment by minsky91** [comment, marked as answer]
The plugin is particularly good at this kind of job, actually. With the Polygon selection tool, trace an area (as precisely as you can, zooming in 200%, for better precision) on the face where you want the mask to be and then use Refine with the Strength sufficiently high; this might require some experimenting before anything materializes. It goes without saying that you have to have a well-written prompt describing the desired mask's look in the prompt box, when generating. When you have a more or less suitable mask generated, use Krita's Selection / Feather to specify the width of the blending border area, about few pixels wide, Invert Selection and Clear with the generated y the plugin layer active; this will soften the transition between the generated content and the face skin. Alternatively to Feather, use plugin's internal Diffusion masking options before generating (though I personally prefer to avoid them). 

With too high Strength values (like, 60% and above), the generated mask can appear a bit out of place on the face, I imagine, so an additional technique you could use in that case to mitigate the effect, is to Refine the newly generated content having the layer created by the plugin partially transparent: the refined content may fit/blend better.  Refine (Custom) is another  option to experiment with generating a new content into the selected area; you could try using its various sub-options for the desired effect. Add Content can be also used, but in my experience, it was rarely useful for this kind of job; Generate/Fill have never worked well for me, for such a purpose.

This is how I imagine I would approach the task, anyway. The plugin's toolset is very rich and versatile, so you might end up using something else. Paradoxically, some great results in generating new content into the selected area can be achieved using the Remove Content plugin's option - again, with a well-traced selection area and a properly written prompt.

**Reply by devzaak** [reply]
Hi Valery, thanks for the ideas, I will try and see if I can work out some ideas to get that to work. Do you perhaps have any recommendation for Lora models that are particularly good at this kind of thing for variety etc? Thanks

**Reply by minsky91** [reply]
Sorry, haven't used any Loras with inpainting, yet (checkpoints, on the other hand, a plenty). Curious about your future experience with them though.
